probatus/feature_elimination/feature_elimination.py

```diff
@@ -389,11 +414,14 @@ def fit(self, X, y, column_names=None):
             scores_train = [current_result[1] for current_result in results_per_fold]
             scores_val = [current_result[2] for current_result in results_per_fold]
 
+            #Calculate the shap features with remaining features and features to keep.
+            remaining_features = list(set(current_features_set) | set(columns_to_keep))
```

Comment:
I would rename it to remaining_removable_features

Reply:
I am also not sure if you should do this here, If you pass it to calculate_shap_importance while in shap values still have these features, this might cause a bug. Since you do not remove these features while calculating the shap values, updating this list like that might cause an issue. 

Please check and/or test it.

Reply:
The remaining_removable_features (RRF) features here contain both the columns_to_keep and other features which have not been eliminated yet. The SHAP values will be calculated on this entire feature set (RRF).  Once the SHAP values are calculated, we do not consider column_to_keep for elimination. Remaining features are eliminated as mentioned in the step parameter. This allows us to calculate the SHAP for the entire RRF that includes the columns_to_keep as well, hence the effect of those features on other feature as well.

Reply:
Nice, that is the desired behaviour